### DISCUSSION TITLE
Pinecil V2 possible C245 modification

### DISCUSSION THREAD

**Discussion by alextrical** [Ralim/IronOS#1696] [Q&A]
Does the SGM8557 / U11 in the Pinecil V2 have the ability to determine the thermistor/temperature sensor value irrespectively of the power being delivered by the HSM4313.
I'm assuming that the 39K pullup resistor R35 is part of the resistor divider being used for the lookup.

If this is the case, It looks like it could be possible to adapt the Pinecil to drive a C245 with separated Power and Sense lines

**Comment by Ralim** [comment]
So you could modify it to split these apart. The only reason we measure the thermocouple out of phase to heat is the combined connections for heat and measurement.

Alternatively most likely you can just merge heater and sense and just put like 100K inline with the sense line (so no power flows through it).

**Comment by alextrical** [comment]
It is possible to use the thermistor inline with the heater in those tips, though it does feel a bit wrong as it isn't designed to handle that level of current (unlike a T12 style tip) 

It can be done, as can be seen by Marco Reps in this video https://www.youtube.com/watch?v=M-EaxWU70xM
Though I believe with his setup the outer shell isn't at GND potential. Splitting the power from the sense could be the nicest solution, and I think that a simple daughter board inside the pinecil could achieve the connection nicely

![JBC Soldering Iron Cartridge Pinout](https://github.com/Ralim/IronOS/assets/35117191/c52ff8dc-43cc-4913-a798-1e7a6f199e6d)

https://www.eevblog.com/forum/testgear/jbc-soldering-station-cd-2bc-complete-schematic-analysis/25/?PHPSESSID=r3tr6qa1541bvnibrdl4ec5pom

**Comment by joelbi** [comment]
This is something I'm potentially interested in. 
Due to the different thermocouple in the c245 cartidges it might be beneficial to change the opamp gain with restistor r25/r27, I dont know how close it would get but with the higher uV/C rating we might reach the limit. 
Is there a high res picture of the PCB somewhere? Being able to cut/ reroute the trace to the opamp to wire up the third contacte of the cartride would be quite beneficial.

**Reply by alextrical** [reply]
I will start looking into the options we have with off the shelf components on a custom PCB carrier (I'm thinking similar to the good old mod-chips) 

I hadn't considered the opamp gain, but that's a very good point, it will likely be needed.
The PCB traces can be found via Pine's documentation here 
https://files.pine64.org/doc/Pinecil/Pinecil_PCB_placement_v2.0_20220608.pdf

For splitting the Power delivery out from the Thermal sense, I'm currently thinking lifting the 2 legs from the MOSSFET would be the easiest route, or if we want to go crazy, maybe relocate the MOSSFET onto the "modchip" style board, and contact the pads on the Pinecil with Castellations

**Reply by joelbi** [reply]
I took a look at the schematic, the easiest way for the thermocouple I see at the moment is the following. The input to the opamp is trough r27. Unsoldering r27 from the pad connecting to A and rerouting it to the 3rd cartidge connection should suffice. Would you like to set up a way to communicate/work on the mod together, that way we wont spam this discussion with potentially irrelevant messages?

**Reply by alextrical** [reply]
That sounds like a good plan, it could make for a easier user mod, assuming the resistor is large enough to hand solder.
Though r27 being on the top side of the board would require a flying lead. It may then requite a 0.1uf cap to replace the smoothing that would be missing from C35 (assuming that would still be required) alongside a 100k resistor

Setting up a communication sounds good to me, are you on Discord or Matrix? Or would a General discussion/channel here be a better choice?

**Reply by joelbi** [reply]
Discord sounds great to me, my username is bidi533
